File: NetDeck/CardCell.m
//
//  CardCell.m
//  Net Deck
//
//  Created by Gereon Steffens on 27.01.14.
//  Copyright © 2016 Gereon Steffens. All rights reserved.
//

#import "CardCell.h"
#import "DeckListViewController.h"

@implementation CardCell

-(void) copiesChanged:(UIStepper*)sender
{
    int copies = sender.value;
    int diff = ABS((int)self.cardCounter.count - copies);
    if (copies < self.cardCounter.count)
    {
        [self.deck addCard:self.cardCounter.card copies:-diff];
    }
    else
    {
        [self.deck addCard:self.cardCounter.card copies:diff];
    }
    self.cardCounter.count = copies;
    
    [[NSNotificationCenter defaultCenter] postNotificationName:Notifications.DECK_CHANGED object:self];
}

-(void) awakeFromNib
{
    [self.identityButton setTitle:l10n(@"Switch Identity") forState:UIControlStateNormal];
}

-(void) selectIdentity:(id)sender
{
    [self.delegate selectIdentity:sender];
}

@end

File: NetDeck/CardCell.swift
//
//  CardCell.swift
//  NetDeck
//
//  Created by Gereon Steffens on 31.12.16.
//  Copyright © 2017 Gereon Steffens. All rights reserved.
//

import UIKit

protocol IdentitySelector: class {
    func selectIdentity(_ sender: Any)
}

class CardCell: UITableViewCell {

    @IBOutlet weak var name: UILabel!
    @IBOutlet weak var influenceLabel: UILabel!
    @IBOutlet weak var copiesStepper: UIStepper!
    @IBOutlet weak var identityButton: UIButton!
    
    weak var delegate: IdentitySelector!
    var deck: Deck!
    var cardCounter: CardCounter? {
        didSet {
            self.setCardCounter(self.cardCounter)
        }
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        self.identityButton.setTitle("Switch Identity", for: .normal)
        self.prepareForReuse()
    }
    
    override func prepareForReuse() {
        super.prepareForReuse()
        
        self.name.text = nil
        self.influenceLabel.text = nil
    }

    @IBAction func selectIdentity(_ sender: Any) {
        self.delegate.selectIdentity(sender)
    }

    @IBAction func copiesChanged(_ stepper: UIStepper) {
        guard let cc = self.cardCounter else { return }
        let count = Int(stepper.value)
        let diff = count - cc.count
        self.deck.addCard(cc.card, copies: diff)
        
        NotificationCenter.default.post(name: Notifications.deckChanged, object: self)
    }
    
    func setCardCounter(_ cc: CardCounter?) {
        let title = cc == nil ? "Choose Identity" : "Switch Identity"
        self.identityButton.setTitle(title.localized(), for: .normal)
        
        if cc == nil {
            self.identityButton.isHidden = false
            self.copiesStepper.isHidden = true
        }
    }
}
